contracts/avalanche/connectors/pangolin/staking/main.sol

```diff
@@ -0,0 +1,133 @@
+
+        _eventName = "LogWithdrawLpStake(address,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(lpTokenAddr, pid, _amt, getId, setId);
+    }
+
+    function withdrawAndClaimLpRewards(
+        uint pid,
+        uint amount,
+        uint256 getId,
+        uint256 setId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        uint _amt = getUint(getId, amount);
+
+        (uint256 rewardAmount, address lpTokenAddr) = _withdraw_and_getRewards_LP_Stake(pid, _amt);
+
+        setUint(setId, _amt);
+
+        _eventName = "LogWithdrawLpAndClaim(address,uint256,uint256,uint256,uint256,uint256)";
+        _eventParam = abi.encode(lpTokenAddr, pid, _amt, rewardAmount, getId, setId);
+    }
+
+    function claimLpRewards(
+        uint pid
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint256 rewardAmount, address lpTokenAddr) = _getLPStakeReward(pid);
+
+        _eventName = "LogClaimLpReward(address,uint256,uint256,)";
+        _eventParam = abi.encode(lpTokenAddr, pid, rewardAmount);
+    }
+
+    function emergencyWithdrawLpStake(
+        uint pid
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint amount, address lpTokenAddr) = _emergencyWithdraw_LP_Stake(pid);
+
+        _eventName = "LogEmergencyWithdrawLpStake(address,uint256,uint256)";
+        _eventParam = abi.encode(lpTokenAddr, pid, amount);
+    }
+
+    // Single Staking
+    function depositPNGStake(
+        address stakingContract,
+        uint256 amount,
+        uint256 getId,
+        uint256 setId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        uint _amt = getUint(getId, amount);
+
+        _depositSingleStake(stakingContract, _amt);
+
+        setUint(setId, _amt);
+
+        _eventName = "LogDepositPNGStake(address,uint256,uint256,uint256)";
+        _eventParam = abi.encode(stakingContract, _amt, getId, setId);
+    }
+
+    function withdrawPNGStake(
+        address stakingContract,
+        uint256 amount,
+        uint256 getId,
+        uint256 setId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        uint _amt = getUint(getId, amount);
+
+        _withdrawSingleStake(stakingContract, _amt);
+
+        setUint(setId, _amt);
+
+        _eventName = "LogWithdrawPNGStake(address,uint256,uint256,uint256)";
+        _eventParam = abi.encode(stakingContract, _amt, getId, setId);
+    }
+
+    function exitPNGStake(
+        address stakingContract
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        (uint256 exitAmount, uint256 rewardAmount, address rewardToken) = _exitSingleStake(stakingContract);
+
+        _eventName = "LogExitPNGStake(address,uint256,uint256,address)";
+        _eventParam = abi.encode(stakingContract, exitAmount, rewardAmount, rewardToken);
+    }
+
+    function getRewardPNGstake(
```

Comment:
camel cased function name here as getRewardPNGStake ?

Reply:
I rename this function to claimPNGStakeReward to look similar to the event and with the function of claimLpRewards